tests/rgen/Microsoft.Macios.Bindings.Analyzer.Tests/SmartEnumsAnalyzerTests.cs

```diff
@@ -0,0 +1,374 @@
+	[Field<EnumValue> (""High"", LibraryName = ""   "")]
+	High,
+}
+";
+
+	[Theory]
+	[AllSupportedPlatforms (customLibraryMissingLibraryName)]
+	[AllSupportedPlatforms (customLibraryEmptyLibraryName)]
+	[AllSupportedPlatforms (customLibraryEmptyLibraryNameParameter)]
+	public async Task SmartEnumThirdPartyLibrary (ApplePlatform platform, string inputText)
+	{
+		var (compilation, _) = CreateCompilation (nameof (SmartEnumsAnalyzerTests), platform, inputText);
+		var diagnostics = await RunAnalyzer (new SmartEnumsAnalyzer (), compilation);
+		var analyzerDiagnotics = diagnostics
+			.Where (d => d.Id == SmartEnumsAnalyzer.RBI0005.Id).ToArray ();
+		Assert.Single (analyzerDiagnotics);
+		VerifyDiagnosticMessage (analyzerDiagnotics [0], SmartEnumsAnalyzer.RBI0005.Id,
+			DiagnosticSeverity.Error, "The enum value 'CustomLibrary.CustomLibraryEnum.High' backing field must provide a library name");
+	}
+
+	[Theory]
+	[AllSupportedPlatforms]
+	public async Task SmartEnumNoFieldAttributes (ApplePlatform platform)
+	{
+		const string inputText = @"
+using System;
+using System.Runtime.Versioning;
+using Foundation;
+using ObjCRuntime;
+using ObjCBindings;
+
+namespace CustomLibrary;
+
+[BindingType]
+public enum CustomLibraryEnum {
+	None,
+	Medium,
+	High,
+}
+";
+		var (compilation, _) = CreateCompilation (nameof (SmartEnumsAnalyzerTests), platform, inputText);
+		var diagnostics = await RunAnalyzer (new SmartEnumsAnalyzer (), compilation);
+		var analyzerDiagnotics = diagnostics
+			.Where (d => d.Id == SmartEnumsAnalyzer.RBI0002.Id).ToArray ();
+		// we should have a diagnostic for each enum value
+		Assert.Equal (3, analyzerDiagnotics.Length);
+	}
+
+	[Theory]
+	[AllSupportedPlatforms]
+	public async Task SmartEnumFieldNotDuplicated (ApplePlatform platform)
+	{
+
+		const string inputText = @"
+using Foundation;
+using ObjCRuntime;
+using ObjCBindings;
+
+namespace AVFoundation;
+
+[BindingType]
+public enum AVCaptureSystemPressureExampleLevel {
+	[Field<EnumValue> (""AVCaptureSystemPressureLevelNominal"")]
+	Nominal,
+
+	// duplicated, this should be an error
+	[Field<EnumValue> (""AVCaptureSystemPressureLevelNominal"")]
+	Fair,
+
+	[Field<EnumValue> (""AVCaptureSystemPressureLevelSerious"")]
+	Serious,
+
+	[Field<EnumValue> (""AVCaptureSystemPressureLevelCritical"")]
+	Critical,
+
+	[Field<EnumValue> (""AVCaptureSystemPressureLevelShutdown"")]
+	Shutdown,
+}";
+
+		var (compilation, _) = CreateCompilation (nameof (SmartEnumsAnalyzerTests), platform, inputText);
+		var diagnostics = await RunAnalyzer (new SmartEnumsAnalyzer (), compilation);
+		var analyzerDiagnotics = diagnostics
+			.Where (d => d.Id == SmartEnumsAnalyzer.RBI0003.Id).ToArray ();
+		Assert.Single (analyzerDiagnotics);
+		VerifyDiagnosticMessage (analyzerDiagnotics [0], SmartEnumsAnalyzer.RBI0003.Id,
+			DiagnosticSeverity.Error, "The enum value 'AVFoundation.AVCaptureSystemPressureExampleLevel.Fair' backing field 'AVCaptureSystemPressureLevelNominal' is already in use");
```

Comment:
> The backing field '...' for the enum value '...' is already in use.

Also, can we provide a diagnostic for the first definition too, or alternatively something like:

> The backing field '...' for the enum value '...' is already in use for the enum value '...'

Reply:
I can do that, move from a hashset to a dict and update the message.